crates/services/p2p/src/discovery/discovery_config.rs

```diff
@@ -167,13 +162,42 @@ impl Config {
             MdnsWrapper::disabled()
         };
 
-        Behaviour {
+        Ok(Behaviour {
             connected_peers: HashSet::new(),
             kademlia,
             next_kad_random_walk,
             duration_to_next_kad: Duration::from_secs(1),
             max_peers_connected,
             mdns,
-        }
+        })
     }
 }
+
+async fn get_kademlia_kv(
+    multiaddrs: Vec<Multiaddr>,
+) -> anyhow::Result<Vec<(PeerId, Multiaddr)>> {
+    let dnsaddr_urls = multiaddrs
+        .iter()
+        .filter_map(|node| {
+            if let Protocol::Dnsaddr(multiaddr) = node.iter().next()? {
+                Some(multiaddr.clone())
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
```

Comment:
Nit: This reads a bit funky to me. Wouldn't the singular of `multiaddrs` be `multiaddr` rather than `node`? I'd suggest this naming instead:

```suggestion
    let dnsaddr_urls = multiaddrs
        .iter()
        .filter_map(|multiaddr| {
            if let Protocol::Dnsaddr(dnsaddr_url) = multiaddr.iter().next()? {
                Some(dnsaddr_url.clone())
            } else {
                None
            }
        })
        .collect::<Vec<_>>();
```

Reply:
addressed in 0482f8023